src/Simulation/Core/IQirMachine.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Quantum.Simulation.Core;
+using Core = Microsoft.Quantum.Simulation.Core;
+
+namespace Microsoft.Quantum.Runtime
+{
+    /// <summary>
+    /// A machine that can execute QIR programs on Azure.
+    /// </summary>
+    public interface IQirMachine : IAzureMachine
+    {
+        /// <summary>
+        /// Submits a job to execute a QIR program without waiting for execution to complete.
+        /// </summary>
+        /// <param name="qir">The QIR program as a byte stream.</param>
+        /// <param name="entryPoint">The fully-qualified name of the entry point to execute.</param>
+        /// <param name="arguments">The arguments to the entry point in the order in which they are declared.</param>
+        /// <returns>The submitted job.</returns>
+        Task<IQuantumMachineJob> SubmitAsync(Stream qir, string entryPoint, IReadOnlyList<Argument> arguments);
+    }
+
+    /// <summary>
+    /// An argument to a QIR callable.
+    /// </summary>
+    public class Argument
```

Comment:
The `Argument` and `ArgumentValue` classes could go in separate files, but there are already a lot of files in this folder... maybe if all of the quantum machine files were in a separate folder it would be more organized?

Reply:
Yes, having the classes in separate files would be good.

I created this other [PR](https://github.com/microsoft/qsharp-runtime/pull/629) moving the rest of the related source files to the _Machines_ folder.

While the PR is merged into the feature branch, you can places these new source files into a _Machines_ and after both PRs are merged we will have things more organized.

Reply:
nit: I know it's become standard terminology for us to refer to these classes as "Machines" because of the naming convention there, but I don't think that makes a very good folder name. It's so generic a term that it's not clear what one would expect to find inside of that folder unless already familiar with our class hierarchy and names. Maybe we can find something more descriptive? At the very least, I think we are getting to the point where the folder hierarchy and how that influences the division of feature implementation is something we should address in on of the readmes (similar to the suggestion from @kuzminrobin regarding more clarity regarding the split between QSharp.Core and QSharp.Foundation).

Reply:
I'm realizing this comment and any further discussion about naming/organization probably better belongs on Cesar's other PR. Please ignore it here, and I'll move over there!

Reply:
> While the PR is merged into the feature branch, you can places these new source files into a Machines and after both PRs are merged we will have things more organized.

@cesarzc, is it right that the argument classes apply not just to submitting a QIR program to Azure, but also to running a QIR program locally if/when the entry point driver supports that? So would it make sense not to put them in the Submitters folder?

Reply:
@samarsha good point. Yes, I think it makes sense to not put them in the _Submitters_ folder.